File: it/structures/csharp/enum-default/Test/EnumImplicit.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.Runtime.Serialization;

namespace Test {
    /// Implicit naming depending on the variant
    [JsonConverter(typeof(StringEnumConverter))]
    public enum EnumImplicit {
        [EnumMember(Value = "A")]
        A,
        [EnumMember(Value = "B")]
        B
    }
}

File: it/structures/java/enum-default/test/EnumImplicit.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Implicit naming depending on the variant
 */
public enum EnumImplicit {
    A("A"),
    B("B");

    String value;

    EnumImplicit(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static EnumImplicit fromValue(final String value) {
        for (final EnumImplicit v : values()) {
            if (v.value.equals(value)) {
                return v;
            }
        }

        throw new IllegalArgumentException("value");
    }

    @JsonValue
    public String toValue() {
        return this.value;
    }
}
